imports/plugins/core/inventory/server/no-meteor/startup.js

```diff
@@ -0,0 +1,221 @@
+          }
+        );
+
+        // Update parents of supplied item
+        await collections.Products.updateMany(
+          {
+            _id: { $in: updatedItem.value.ancestors }
+          },
+          {
+            $inc: {
+              inventoryAvailableToSell: +item.quantity
+            }
+          }
+        );
+      });
+
+      // Publish inventory updates to the Catalog
+      // Since variants share the same productId, we only want to update each product once
+      // So we use a Set to get all unique productIds that were affected, then loop through that data
+      const productIds = [...new Set(orderItems.map((item) => item.productId))];
+      productIds.forEach(async (productId) => {
+        await updateCatalogProductInventoryStatus(productId, collections);
+      });
+    }
+  });
+
+  appEvents.on("afterOrderCreate", async (order) => {
+    const { collections } = context;
+    const orderItems = order.shipping.reduce((list, group) => [...list, ...group.items], []);
+
+    // Create a new set of unique productIds
+    // We do this because variants might have the same productId
+    // and we don't want to update the product each time a variant is it's child
+    // we can map over the unique productIds at the end, and update each one once
+    orderItems.forEach(async (item) => {
+      // Update supplied item inventory
+      const updatedItem = await collections.Products.findOneAndUpdate(
+        {
+          _id: item.variantId
+        },
+        {
+          $inc: {
+            inventoryAvailableToSell: -item.quantity
+          }
+        }
+      );
+
+      // Update supplied item inventory
+      await collections.Products.updateMany(
+        {
+          _id: { $in: updatedItem.value.ancestors }
+        },
+        {
+          $inc: {
+            inventoryAvailableToSell: -item.quantity
+          }
+        }
+      );
+    });
+
+    // Publish inventory updates to the Catalog
+    // Since variants share the same productId, we only want to update each product once
+    // So we use a Set to get all unique productIds that were affected, then loop through that data
+    const productIds = [...new Set(orderItems.map((item) => item.productId))];
+    productIds.forEach(async (productId) => {
+      await updateCatalogProductInventoryStatus(productId, collections);
+    });
+  });
+
+  appEvents.on("afterOrderApprovePayment", async (order) => {
+    const { collections } = context;
+    const orderItems = order.shipping.reduce((list, group) => [...list, ...group.items], []);
+
+    // Create a new set of unique productIds
+    // We do this because variants might have the same productId
+    // and we don't want to update the product each time a variant is it's child
+    // we can map over the unique productIds at the end, and update each one once
+    orderItems.forEach(async (item) => {
+      // Update supplied item inventory
+      const updatedItem = await collections.Products.findOneAndUpdate(
```

Comment:
Unless you set the `returnOriginal` option to `false`, `findOneAndUpdate` default behavior is to return the original, not the updated. In this case it doesn't cause an issue, but `updatedItem` is an inaccurate variable name

Reply:
Also the return is technically an object where `value` is the document, either original or updated. So it would be better to do

```js
const { value: originalProduct } =
```

Reply:
Same comment for the other similar calls in this PR, but it doesn't seem to be anything more than a variable name issue in all places.

Reply:
It doesn't really matter, but do you think it might be best to have the updated document for use in the future, to expand on this, if needed: https://github.com/reactioncommerce/reaction/pull/4859/commits/09e2b4c0a9ace7f442069bf197d9222f1aba318e

Reply:
Agree that's best, but the option is called `returnOriginal` and needs to be set to `false`: http://mongodb.github.io/node-mongodb-native/3.1/api/Collection.html#findOneAndUpdate

Reply:
Got it...updated. Was running off of the MongoDB Shell docs: https://docs.mongodb.com/manual/reference/method/db.collection.findOneAndUpdate/